sdk/src/connections/offline/offline_depth_sensor.cpp

```diff
@@ -35,16 +37,38 @@ aditof::Status OfflineDepthSensor::getFrame(uint16_t *buffer) {
 aditof::Status OfflineDepthSensor::open() {
     std::vector<std::string> frameTypesResources;
     std::streampos size;
-    std::string fileName;
     uint16_t *buffer;
+#ifdef UNIX
+    std::string fileName;
     struct dirent *entry;
     DIR *dir = opendir(m_path.c_str());
     while ((entry = readdir(dir)) != NULL) {
         fileName = entry->d_name;
-        if (fileName != "." && fileName != "..")
+        if (fileName[0] != ".")
             frameTypesResources.push_back(fileName);
     }
-
+#elif WIN32
+    HANDLE dir;
+    WIN32_FIND_DATA file_data;
+    if ((dir = FindFirstFile((m_path + "/*").c_str(), &file_data)) ==
+        INVALID_HANDLE_VALUE)
+        return aditof::Status::UNREACHABLE;
+    do {
+        std::string fileName = file_data.cFileName;
+        const bool is_directory =
+            (file_data.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
+
+        if (fileName[0] == '.')
+            continue;
+
+        if (is_directory)
+            continue;
+
+        frameTypesResources.push_back(fileName);
+    } while (FindNextFile(dir, &file_data));
+
+    FindClose(dir);
+#endif
     for (int index = 0; index < frameTypesResources.size(); index++) {
```

Comment:
Can you check if there any file paths in this variable before the for loop? And throw an error if the path list is empty.

Reply:
Solved